c4_build_catalog.m

```diff
@@ -1,44 +1,67 @@
 
 Cooksey_C4_detected = fitsread('data/C4_catalogs/Cooksey_C4_cat/distfiles/Cooksey_C4_detected.fits', 'binarytable');
-c4_QSO_ID = Cooksey_C4_detected{1};
-c4_NCIV = Cooksey_C4_detected{10};
-c4_zCIV = Cooksey_C4_detected{3};
+c4_QSO_ID           = Cooksey_C4_detected{1};
+c4_NCIV             = Cooksey_C4_detected{10};
+c4_zCIV             = Cooksey_C4_detected{3};
+
 f = fopen('data/C4_catalogs/Cooksey_C4_cat/processed/los_catalog','w');
 for i=1:size(c4_zCIV)
-    
     fprintf(f,'%s \n', c4_QSO_ID{i});
-
 end
 
 f = fopen('data/C4_catalogs/Cooksey_C4_cat/processed/c4_catalog','w');
-
 for i=1:size(c4_zCIV)
-    
-    fprintf(f,'%s  %f %f\n', c4_QSO_ID{i}, c4_zCIV(i), c4_NCIV(i));
-
+      fprintf(f,'%s  %f %f\n', c4_QSO_ID{i}, c4_zCIV(i), c4_NCIV(i));
 end
-% There are some NAN valued c4_NCIV
 
+% There are some NAN valued c4_NCIV
 % extract basic QSO information from Cookse_all_QSO catalog 
-cooksey_all_qso_catalog =  fitsread('data/dr7/distfiles/Cooksey_all_QSO.fits', 'binarytable');
-all_QSO_ID              =  cooksey_all_qso_catalog{1};
+cooksey_all_qso_catalog = fitsread('data/dr7/distfiles/Cooksey_all_QSO.fits', 'binarytable');
+all_QSO_ID              = cooksey_all_qso_catalog{1};
+all_RAh                 = cooksey_all_qso_catalog{2};
+all_RAm                 = cooksey_all_qso_catalog{3};
+all_RAs                 = cooksey_all_qso_catalog{4};
+all_DEC_Sign_d          = cooksey_all_qso_catalog{5};
+all_DEC_m               = cooksey_all_qso_catalog{6};
+all_DEC_s               = cooksey_all_qso_catalog{7};
 all_zqso                = cooksey_all_qso_catalog{8};
-all_snrs                =  cooksey_all_qso_catalog{9};
-bal_visual_flags        = cooksey_all_qso_catalog{10};
+all_snrs                = cooksey_all_qso_catalog{9};
+all_bal_flags           = cooksey_all_qso_catalog{10};
 num_quasars             = numel(all_zqso);
 
+% Converting RA in Hour angle to ordinary degree
+all_ras = all_RAh.*15 + all_RAm./4 + all_RAs./240;
+all_decs = all_RAh.*15 + all_RAm./4 + all_RAs./240;
+
+%  Converting DEC in deg-miin-sec to degree
+for i=1:numel(all_DEC_Sign_d)
+    if(all_DEC_Sign_d(i)>0)
+        all_decs(i) = all_DEC_Sign_d(i) + all_DEC_m(i)./60 + ...
+            all_DEC_s(i)./3600;
+    else
+        all_decs(i) = all_DEC_Sign_d(i) - all_DEC_m(i)./60 - ...
+            all_DEC_s(i)./3600;
+    end
+end
+
+% save catalog 
+release = 'dr7';
+variables_to_save = {'all_QSO_ID', 'all_ras', 'all_decs', 'all_zqso',...
+    'all_snrs', 'all_bal_flags'};
```

Comment:
I think `filter_flags` should also be saved?

Reply:
I think I cannot save filter_flag here, because it is build in c4_preload_qso.m? 
I saved filter_flag in the c4_set_parameter after it is build. in line 90
```
% write new filter flags to catalog
save(sprintf('%s/catalog', processed_directory(release)), ...
     'filter_flags', '-append');
```

Reply:
Oh that makes sense!